articles/tutorials/qft-tutorial.md

```diff
@@ -0,0 +1,624 @@
+
+In addition, we will use the [`DumpMachine`](xref:microsoft.quantum.diagnostics.dumpmachine) function to observe how the simulated wavefunction of our three qubit system evolves across the operation.
+
+To create a your Q# project, follow these [instructions](xref:microsoft.quantum.howto.createproject), and create a Q# file entitled `Operations.qs`.
+We will walk you through the components of the file step-by-step, but the code is also available as a full block below.
+
+### Namespaces to access other Q# operations
+Inside the file, we first define the namespace `Quantum.Operations` which will be accessed/imported by the classical host.
+For our operation to make use of existing Q# operations, we open the relevant `Microsoft.Quantum.<>` namespaces.
+
+```qsharp
+namespace Quantum.Operations {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Diagnostics;
+	open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Arrays;
+
+	// operations go here
+}
+```
+
+### Define operations with arguments and returns
+Next, we define the `Perform3qubitQFT` operation:
+
+```qsharp
+    operation Perform3qubitQFT() : Unit {
+		// do stuff
+	}
+```
+For now, the operation takes no arguments and does not return anything---in this case we write that it returns a `Unit` object, which is akin to `void` in C# or an empty tuple, `Tuple[()]`, in Python.
+Later, we will modify it to return an array of measurement results, at which point `Unit` will be replaced by `Result[]`. 
+
+### Allocate qubits with `using`
+Within our Q# operation, we first allocate a register of three qubits with the `using` statement:
+
+```qsharp
+        using (qs = Qubit[3]) {
+
+            Message("Initial state |000>:");
+            DumpMachine();
+
+		}
+```
+
+With `using`, the qubits are automatically allocated in the $\ket{0}$ state. We can verify this by using [`Message(<string>)`](xref:microsoft.quantum.intrinsic.message) and [`DumpMachine()`](xref:microsoft.quantum.diagnostics.dumpmachine), which print a string and the system's current state to the host console.
+
+> [!NOTE]
+> The `Message(<string>)` and `DumpMachine()` functions (from [`Microsoft.Quantum.Intrinsic`](xref:microsoft.quantum.intrinsic) and [`Microsoft.Quantum.Diagnostics`](xref:microsoft.quantum.diagnostics), respectively) both print directly to our classical host's console. 
+> Just like a real quantum computation, Q# does not allow us to directly access qubit states.
+> However, as `DumpMachine` prints the target machine's current state, it can provide valuable insight for debugging and learning when used in conjunction with the full state simulator.
+
+
+### Applying single-qubit and controlled gates
+
+Next, we apply the gates which comprise the operation itself (see the circuit diagram and explanation for the three-qubit example [here](https://en.wikipedia.org/wiki/Quantum_Fourier_transform#Example)).
```

Comment:
Although we ask learners read through the wikipedia page of QFT, it may still be good that we provide the circuit diagram as we construct the circuit in Q# step-by-step. 

![image](https://user-images.githubusercontent.com/59890648/76322362-94b1cd00-62e3-11ea-9064-b8a42ce96cbe.png)

Reply:
If possible, I would suggest against including a circuit diagram to avoid furthering the existing impression that Q# is a circuit-generation tool rather than a quantum programming language.

Reply:
Understand the concern. But as a self-contained tutorial, with this one specifically about programming each qubit, it gives better experience for readers to know where the code is at.